core/src/main/java/com/linecorp/armeria/common/util/ShutdownHooks.java

```diff
@@ -87,6 +87,7 @@ private static CompletableFuture<Void> addClosingTask(
                 closeFuture.completeExceptionally(cause);
             }
         };
+        // todo(BueVonHun): in this case, it has some design options...
```

Comment:
It would not be invoked during request processing. That being said, we may add a lock as a static variable so that we can ensure `synchronized` isn't used in our code base.

Reply:
I sent out commit that Implemented lock as a static variable🙂